tutorials/advanced-topics/lazy-propagation.md

```diff
@@ -0,0 +1,189 @@
+---
+title: "Lazy Propagation"
+
+description: "Author: @Dhruvil-Lakhtaria"
+
+hide_table_of_contents: true
+---
+
+<TutorialAuthors  names="@Dhruvil-Lakhtaria"/>
+
+### Introduction
+
+Lazy Propagation comes into picture with Segment Tree when the operation is no longer point update.Instead we want to do a range update. That is instead of updating a single value we are updating a $$range(L - R)$$ with a value.
+
+If we use our standard Segment Tree we can do the update operation for each element in $$[L - R]$$.
+But this will make the time complexity of range update as $$O(NlogN)$$.
+
+### Idea behind Lazy Propagation
+
+The idea behind the algorithm is don’t not update a node until needed which will avoid the repeated sharing.
+
+Updating the interval represented by a node is similar to update the subtree of that node in segment tree. Suppose we have to update subtree of **Node X**, instead of updating all the values at that instant we will only update that node and mark its children that their subtree has to be updated and for maintaining this record we will use **Lazy array**. Size of Lazy array would be same as the size of segment tree. We will initialize the Lazy array with **0** which indicates there is no pending updates. While processing a node, let say **u**, if **Lazy[u] is non-zero**, it implies there is a pending update so before doing any further operation we will first update node u and if its children exist make them lazy.
```

Comment:
https://medium.com/nybles/understanding-range-queries-and-updates-segment-tree-lazy-propagation-and-mos-algorithm-d2cd2f6586d8

Another copycat

Reply:
I referred to this blog for learning and kept the same content for my blog.

Reply:
https://github.com/wingkwong/leetcode-the-hard-way/blob/main/.github/pull_request_template.md#checklist

>  This Pull Request is all my own work. (You'll be blacklisted if you are caught for plagiarism.)

Reply:
I just copied the theory part rest solution and all are mine only